src/uct/ib/ud/base/ud_iface.c

```diff
@@ -234,7 +234,7 @@ static void uct_ud_iface_async_handler_remove(ucs_callbackq_slow_elem_t *self)
 
     ucs_trace_func("iface=%p remove async fd=%d and unregister. slow-path %d",
                    iface, iface->super.comp_channel->fd, iface->async.cbq_elem_on);
-    ucs_async_unset_event_handler(iface->super.comp_channel->fd);
+    ucs_async_remove_handler(iface->super.comp_channel->fd, 1);
```

Comment:
see my not above - asyn can also be on same thread, so doing as you suggest here would add even more confusion